### PR TITLE
gh-92417: `importlib` docs: remove references to unsupported Python versions

### PR DESCRIPTION
**Pull request by AlexWaygood**
#92417

### CODE REVIEW

File: Doc/library/importlib.rst [hunk 1]
```diff
@@ -1355,8 +1354,7 @@ Import itself is implemented in Python code, making it possible to
 expose most of the import machinery through importlib. The following
 helps illustrate the various APIs that importlib exposes by providing an
 approximate implementation of
-:func:`importlib.import_module` (Python 3.4 and newer for the importlib usage,
-Python 3.6 and newer for other parts of the code).
+:func:`importlib.import_module`.
 ::
```
Comment (CAM-Gerlach) [inline Doc/library/importlib.rst]:
```suggestion
:func:`importlib.import_module`::
```

Minor nit (present in the original): Flow is a bit cleaner and more idiomatic here with a colon after a "The following [...] <illustrates> <thing>"-patterned phrase to explicitly indicate the connection and match the usage here and elsewhere.

**Review by CAM-Gerlach — APPROVED** [review]
Aside from one minor textual nit (that wasn't introduced by this PR, but on a line touched by it), I checked over the importlib* docs and there indeed doesn't appear to be remaining references to unsupported versions, so LGTM.

One note, though—will only changes referring to Python versions that were unsupported at the time the backport target was released be backported? It seems a little odd to actively remove docs content that refers to versions supported at the time an older bugfix version just because an EoL date was later reached, as well as extra work.

It would seem to make more sense to have a consistent policy of just mentioning the Python versions supported as of the release of the target version, rather than going through and updating all the bugfix, pre-release and future docs at the moment an older version goes EoL, since otherwise up to 4 sets of docs need to be updated with every EoL on a rigid schedule, as opposed to just one set per release at any point between branching the previous release and the current release.

File: Doc/library/importlib.rst (hunk 1 above)
Comment (AlexWaygood) [reply Doc/library/importlib.rst]: I generally prefer to err on the side of keeping the diff as small as possible, but I also agree that this does improve the flow.

**Comment by jaraco** [comment]
This change also needs to get backported to importlib_metadata.

**Pull request merged**